### ISSUE TITLE
exception on node shutdown

### ISSUE THREAD

**Issue by pdeva** [apache/incubator-druid#5568]
Happens in 0.12.0, but i am sure i noticed in 0.11.0 too

```
2018-04-03 19:40:38,809 Thread-49 ERROR Unable to register shutdown hook because JVM is shutting down. java.lang.IllegalStateException: Not started
	at io.druid.common.config.Log4jShutdown.addShutdownCallback(Log4jShutdown.java:47)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.addShutdownCallback(Log4jContextFactory.java:273)
	at org.apache.logging.log4j.core.LoggerContext.setUpShutdownHook(LoggerContext.java:256)
	at org.apache.logging.log4j.core.LoggerContext.start(LoggerContext.java:216)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:145)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:41)
	at org.apache.logging.log4j.LogManager.getContext(LogManager.java:182)
	at org.apache.logging.log4j.spi.AbstractLoggerAdapter.getContext(AbstractLoggerAdapter.java:103)
	at org.apache.logging.slf4j.Log4jLoggerFactory.getContext(Log4jLoggerFactory.java:43)
	at org.apache.logging.log4j.spi.AbstractLoggerAdapter.getLogger(AbstractLoggerAdapter.java:42)
	at org.apache.logging.slf4j.Log4jLoggerFactory.getLogger(Log4jLoggerFactory.java:29)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:253)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:265)
	at org.apache.curator.RetryLoop.<init>(RetryLoop.java:65)
	at org.apache.curator.CuratorZookeeperClient.newRetryLoop(CuratorZookeeperClient.java:151)
	at org.apache.curator.connection.StandardConnectionHandlingPolicy.callWithRetry(StandardConnectionHandlingPolicy.java:59)
	at org.apache.curator.RetryLoop.callWithRetry(RetryLoop.java:100)
	at org.apache.curator.framework.imps.CuratorTransactionImpl.commit(CuratorTransactionImpl.java:123)
	at io.druid.curator.announcement.Announcer.stop(Announcer.java:170)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.druid.java.util.common.lifecycle.Lifecycle$AnnotationBasedHandler.stop(Lifecycle.java:434)
	at io.druid.java.util.common.lifecycle.Lifecycle.stop(Lifecycle.java:335)
	at io.druid.java.util.common.lifecycle.Lifecycle$1.run(Lifecycle.java:366)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by clintropolis** [comment]
This should be fixed by #6864.

**Comment by shaoxinye** [comment]
ERROR Unable to register shutdown hook because JVM is shutting down. java.lang.IllegalStateException: Not started

who can help me？
